PlainSql.Migrations.Generator/Program.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+using McMaster.Extensions.CommandLineUtils;
+using Serilog;
+
+namespace PlainSql.Migrations.Generator;
+
+class Program
+{
+    public static int Main(string[] args) => CommandLineApplication.Execute<Program>(args);
+
+    private const string Placeholder = "SELECT * FROM Students WHERE Name = 'Robert'); DROP TABLE Students;--";
+
+    [Required]
+    [Argument(0)]
+    private string Description { get; }
+
+    [Option(Description = "The folder where the migration scripts are located",
+        ShortName = "m",
+        LongName = "migrationscriptfolder")]
+    private string MigrationScriptFolder { get; set; } = "./MigrationScripts";
+
+    private void OnExecute()
+    {
+        Log.Logger = new LoggerConfiguration()
+            .WriteTo.Console()
+            .CreateLogger();
+
+        try
+        {
+            CheckIfMigrationScriptFolderExists();
+
+            var filename = CreateFilename();
+
+            Log.Information("Creating migration file {filename} in {MigrationScriptFolder}",
+                filename,
+                MigrationScriptFolder);
+
+            CreateMigrationFile(filename);
+
+            Log.Information("Sucessfully created {filename}!", filename);
+        }
+        catch (Exception e)
+        {
+            Log.Error("There has been a problem creating the migration file: " + e.Message);
```

Comment:
Any particular reason why the Exception `e` is not provided as the first parameter in `Log.Error()`?

Reply:
Yes, the output is way cleaner. But I wonder if it's worth to complicate troubleshooting in case of other errors (although they might be highly unlikely...)

Anyway I don't mind either way, was just interested :-)

Reply:
You're right of course, error messages are the wrong place to worry more about nice looking ouput than usable information. I've changed it.